tests/common/mock/mock_serial_port.rs

```diff
@@ -21,7 +21,7 @@ impl SharedBuffer {
 		}
 	}
 
-	pub fn read(&self) -> Option<MutexGuard<Vec<u8>>> {
+	pub fn read<'a>(&'a self) -> Option<MutexGuard<'a, Vec<u8>>> {
```

Comment:
```suggestion
	pub fn read(&self) -> Option<MutexGuard<'_, Vec<u8>>> {
```

Small note: this is the more idiomatic way to write this. But I don't mind seeing the explicit lifetimes, so let's keep them :)